src/drf_pydantic/parse.py

```diff
@@ -243,15 +245,14 @@ def _convert_type(  # noqa: PLR0911
 
     # Scalar field
     if is_scalar(type_):
-        # Nested model
-        if issubclass(type_, pydantic.BaseModel):
-            try:
-                return getattr(type_, "drf_serializer")(**kwargs)
-            except AttributeError:
-                return create_serializer_from_model(type_)(**kwargs)
-
-        # Normal class
         if inspect.isclass(type_):
```

Comment:
As I think about it, we don't need this check at all - `type_` must be a class, pydantic wouldn't allow anything else (and if it does, then it's a bug). Please revert this portion and I'll refactor this code block in future release.

Reply:
Pydantic supports not only classes but `TypeAliasType` instances as well, which are `inspect.isclass == False`. 
`JsonValue` is one of the examples. The logic I was trying to achieve here is:
- First level of identification is scalar/composite (defined explicitly)
- Second level - special cases (submodels, Decimals, Enums, etc.) involving calling of `is_subclass`. We need to ensure we deal with a class before calling `is_subclass` because it will raise `TypeError` otherwise.
- Third level - fallback to explicitly defined FIELD_MAP, where keys could be any types including aliases.

Reply:
Can you please create a test scenario which results in such `TypeError`? I'd like to see what we are dealing with before handling that.

Reply:
I believe it falls outside of this PR scope. Can you please post an example of model which would cause this? Let's create another PR for that issue and merge this one for `JSONField` only.

Reply:
Not sure I completely understand your request :) 
The `TypeError` is fixed in this PR by putting all `is_subclass` calls under `if if inspect.isclass(type_):`. If I would leave the logic unchanged - JsonValue wouldn't work.
![drf-pydantic 2024-02-06 11-28-58](https://github.com/georgebv/drf-pydantic/assets/11632032/121fff92-195c-450f-a553-9e6055d72493)

Reply:
@georgebv lets merge it?

Reply:
@djbios thank you for explaining this, I see what's going on now